exec/java-exec/src/main/java/org/apache/drill/exec/planner/FileSystemPartitionDescriptor.java

```diff
@@ -157,7 +159,20 @@ private String getBaseTableLocation() {
 
   @Override
   protected void createPartitionSublists() {
-    List<String> fileLocations = ((FormatSelection) table.getSelection()).getAsFiles();
+    Collection<String> fileLocations = null;
+    if (scanRel instanceof DrillScanRel) {
```

Comment:
I will try to create an interface at the GroupScan level to make this transparent.